src/logics/Logic.cc

```diff
@@ -851,8 +851,17 @@ PTRef Logic::insertTerm(SymRef sym, vec<PTRef>&& terms)
 PTRef
 Logic::mkFun(SymRef sym, vec<PTRef>&& terms)
 {
+    assert(
+        [this](SymRef sym, vec<PTRef> const & terms) {
+            std::string why;
+            bool rval = typeCheck(sym, terms, why);
+            if (not rval) {
+                throw OsmtApiException(why);
+            }
+            return rval;
+        }(sym, terms)
+    );
```

Comment:
Somehow, this `assert` does not feel like the best way, since it can never evaluate to false. It will either return true or throw an exception.
If you really want to throw an exception here, maybe we can just wrap this in `ifdef`?
```
#ifndef NDEBUG
std::string why;
if (not typeCheck(sym, terms, why)) {
   throw OsmtApiException(why);
}
#endif
```

Reply:
That sounds good.  I think it makes sense for now to throw until we figure out the way we want to type checking to happen and when.  Maybe an internal exception would make more sense for now?

Reply:
I am fine with either case.